v0.7.0 Pre-Release: More config options, LDAP/remote auth support, extractor updates, usability and performance improvements, bugfixes

pirate opened the pull request:
DONE:

- [x] pass `--single-process --no-zygote` args to chrome in Docker to mitigate orphan subprocess accumulation issues (also made it a lot faster as a side-effect!) 49faec8f6
- [x] fix user and python binary detection config for FreeBSD 208f866, 831d82a7
- [x] update automatic setup script to install via Docker/Docker-compose if possible, fall back to apt/brew/pkg/pip 06f6084e3
- [x] only warn on rejected stdin instead of throwing an error 06f6084e
- [x] fix bookmarklet not uriencoding urls b6ee7d291
- [x] throw explicit error when django version is too low 9563e74f
- [x] always try to install all dependencies on `archivebox setup` instead of trying to detect `USE_` config options 3e7d8e52
- [x] exempt `/add` from CSRF checking in order to support submissions from ArchiveBox browser extensions https://github.com/ArchiveBox/ArchiveBox/pull/777
- [x] add support for new `URL_WHITELIST` option to restrict added URLs to ones matching a given regex pattern 5a2c78e https://github.com/ArchiveBox/ArchiveBox/wiki/Configuration#URL_WHITELIST
- [x] #658 fix admin panel actions in safari on iOS/macOS
- [x] #843 fix KeyError during some Pocket API import parses
- [x] #901
- [x] https://github.com/ArchiveBox/ArchiveBox/issues/971 Fix wallabag CSV parser acd53c8
- [x] lots of minor UI and CLI fixes
- [x] #847
- [x] #22 bump Sonic version to support ARM
- [x] https://github.com/ArchiveBox/readability-extractor/issues/2 bump `readability-extractor` version to `v0.0.3` to fix relative URL rewriting
- [x] #645 add option to disable original preview / all preview iframes on snapshot index pages
- [x] #944 Fix broken URLs blocking entire depth=1 imports
- [x] #706 fix bytes string being passed to hint causing exception
- [x] #725 fix import tags being split into single characters and emptystring/whitespace-only tags
- [x] #734 expose new `ARCHIVEBOX_USERNAME=xyz` and `ARCHIVEBOX_PASSWORD=xyz` env variables to make initial docker setup easier (so that manual `init` / `manage createsuperuser` isn't required as a separate setup)
- [x] #1002 fix permissions/migration creation errors
- [x] #1036 fix stdin/stdout handling edge case when using nohup
- https://github.com/ArchiveBox/ArchiveBox/pull/1249
- https://github.com/ArchiveBox/ArchiveBox/pull/1245
- https://github.com/ArchiveBox/ArchiveBox/pull/1242
- https://github.com/ArchiveBox/ArchiveBox/pull/1168
- https://github.com/ArchiveBox/ArchiveBox/pull/996
- https://github.com/ArchiveBox/ArchiveBox/pull/1214
- https://github.com/ArchiveBox/ArchiveBox/pull/1221
- https://github.com/ArchiveBox/ArchiveBox/pull/1219
- https://github.com/ArchiveBox/ArchiveBox/pull/1211
- https://github.com/ArchiveBox/ArchiveBox/pull/911
- https://github.com/ArchiveBox/ArchiveBox/pull/1199
- https://github.com/ArchiveBox/ArchiveBox/pull/1197
- https://github.com/ArchiveBox/ArchiveBox/pull/1194
- https://github.com/ArchiveBox/ArchiveBox/pull/1154
- https://github.com/ArchiveBox/ArchiveBox/pull/1146
- https://github.com/ArchiveBox/ArchiveBox/pull/1026
- https://github.com/ArchiveBox/ArchiveBox/pull/1115
- https://github.com/ArchiveBox/ArchiveBox/pull/1068
- https://github.com/ArchiveBox/ArchiveBox/pull/866
- https://github.com/ArchiveBox/ArchiveBox/pull/1021
- https://github.com/ArchiveBox/ArchiveBox/pull/528
- https://github.com/ArchiveBox/ArchiveBox/pull/967
- https://github.com/ArchiveBox/ArchiveBox/pull/1241

TODO:


Postpone:
- [ ] #696 show prompt to upgrade archivebox in UI when new version is available
- [ ] #761 add support to `download_url` to use cookies in `COOKIES_FILE` (used for title fetching and other things)
- [ ] #726 fix pocket `since` parameter preventing failed depth=1 import from resuming
- [ ] #785 fix Pinboard import not respecting original bookmarked timestamps
- [ ] fix `oneshot` command broken: `archivebox oneshot --extract=singlefile,readability https://example.com` -> `FileNotFoundError in log_link_archiving_finished(): [Errno 2] No such file or directory: '.../archive/1620881375.32482'`
- [ ] #783 fix sorting by size column in Snapshot admin causing 500 errors

peterrus commented:
Hi @pirate, It seems the Wallabag fix does not work as intended: https://github.com/ArchiveBox/ArchiveBox/issues/971#issuecomment-1122499507

Just tried it again with `archivebox/archivebox:sha-03eb7e5` and can confirm this is still an issue (for me at least).

mAAdhaTTah commented:
Build failed, possibly due to recent CloudFlare issues: https://www.cloudflarestatus.com/incidents/l6x2h1zp69bc

pirate commented:
FYI everyone watching this release, I just fixed the build, among many other things. Give it a try!

> My apologies, just fixed the build and just pushed `dev`, (aka `0.7.0`, aka `latest`) for `amd64`, `arm64`, and `arm/v7` (run `./bin/build_docker.sh` to build these yourself).
> 
> * fix issue where error output was writing to `>./2` instead of `>&2`
> * massively increased the speed of all the Docker builds (>20x) by adding granular shared cache mounts for reusable apt and pip caches ([`--mount=type=cache...`](https://github.com/docker/buildx/issues/549#issuecomment-1788266709))
> * fixed multi-arch docker builds for `amd64`, `arm64`, `arm/v7` and updated Dockerfile and `./bin/build_docker.sh` with latest config and `PDM` package manager
> * attempt to autodetect PUID and PGID based on the ownership of existing files within `./data`, should prevent unecessary `chown`s and remove the need to pass `PUID`/`PGID` explicitly in many cases
> * and added more helpful instructions to clearly show when a user has hit an edge case that requires passing explicit PUID & PGID (like non-compliant NFS/Samba/FUSE mounts or Docker Desktop on some platforms)
> * updated the base image to `python:3.11-bookworm-slim` with `bookworm-backports`, `node v21`, `yt-dlp 2023.10.13`, `chrome v119`, and all other `apt`, `pip`, and `npm` dependency versions upgraded
> * with a lot of [research, yak-shaving, and trial and error](https://github.com/ArchiveBox/ArchiveBox/commits/dev/) I've managed to get the Docker image down to [500~630mb for all platforms](https://hub.docker.com/r/archivebox/archivebox/tags) (it was >1GB in the previous version), but I cant promise it'll stay this small forever. It's a lot of work to keep it as small as possible for all architectures!
> * added new `/VERSION.txt` file containing build summary, check it on your platform for lots of juicy details about what version you're running
> 
> ```shell
> # pull the :latest image from docker hub (also published to ghcr.io/archivebox/archivebox)
> docker pull archivebox/archivebox
> 
> # get detailed info about the build and dependency versions
> docker run archivebox/archivebox cat /VERSION.txt
> docker run -it archivebox/archivebox version
> 
> # run update to retry any previously failed downloads using the latest extractor versions
> docker run -it -v $PWD:/data archivebox/archivebox update
> ```
> 
> [hub.docker.com/layers/archivebox/archivebox/dev/images/sha256-afa301d566a01cab8934aa1141b5f6133a4585ee5b449304c659c765e240575d?context=explore](https://hub.docker.com/layers/archivebox/archivebox/dev/images/sha256-afa301d566a01cab8934aa1141b5f6133a4585ee5b449304c659c765e240575d?context=explore)

https://github.com/ArchiveBox/ArchiveBox/issues/1247#issuecomment-1788269076
